mbedtls/src/error.rs

```diff
@@ -23,66 +24,114 @@ pub trait IntoResult: Sized {
     }
 }
 
+pub mod codes {
+    pub use super::HiError::*;
+    pub use super::LoError::*;
+}
+
 // This is intended not to overlap with mbedtls error codes. Utf8Error is
 // generated in the bindings when converting to rust UTF-8 strings. Only in rare
 // circumstances (callbacks from mbedtls to rust) do we need to pass a Utf8Error
 // back in to mbedtls.
 pub const ERR_UTF8_INVALID: c_int = -0x10000;
 
 macro_rules! error_enum {
-    {enum $n:ident {$($rust:ident = $c:ident,)*}} => {
-        #[derive(Debug, Eq, PartialEq)]
+    {
+        const MASK: c_int = $mask:literal;
+        enum $n:ident {$($rust:ident = $c:ident,)*}
+    } => {
+        #[non_exhaustive]
+        #[derive(Debug, Eq, PartialEq, Copy, Clone)]
         pub enum $n {
             $($rust,)*
-            Other(c_int),
-            Utf8Error(Option<Utf8Error>),
-            // Stable-Rust equivalent of `#[non_exhaustive]` attribute. This
-            // value should never be used by users of this crate!
-            #[doc(hidden)]
-            __Nonexhaustive,
+            Unknown(c_int)
         }
 
-        impl IntoResult for c_int {
-            fn into_result(self) -> Result<c_int> {
-                let err_code = match self {
-                    _ if self >= 0 => return Ok(self),
-                    ERR_UTF8_INVALID => return Err(Error::Utf8Error(None)),
-                    _ => -self,
-                };
-                let (high_level_code, low_level_code) = (err_code & 0xFF80, err_code & 0x7F);
-                Err($n::from_mbedtls_code(if high_level_code > 0 { -high_level_code } else { -low_level_code }))
+        impl From<c_int> for $n {
+            fn from(code: c_int) -> $n {
+                // check against mask here (not in match blook) to make it compile-time
```

Comment:
nit: `blook` -> `block`

Reply:
(I understand that this comes from the original PR, though)

Reply:
addressed this. `blook -> block`